### DISCUSSION TITLE
Row Count

### DISCUSSION THREAD

**Discussion by AGlennDa** [trimble-oss/dba-dash#832] [Q&A] [answered]
Is there a way in DBA Dash to monitor row counts? 
I am working in a production environment, where we have had luck with monitoring row counts. 
(Until now, we have had our own developed application, just like DBA Dash, but without all the features :) )
Anyway. Our application were monitoring the row count. And then a threshold was set, that we could then expand on if needed. 
So eg.: 
dbo.table1 has a threshold of a random row, with a count on 50.000 rows. 
If the row count this morning from the last collection was above 50.000 rows, we would get an alert in our dashboard. 
We could then investigate if something was spamming the table, or if some real data was put into the row, and we could then expand the threshold to lets say 55.000 rows. 
In a production MES system, rows should not grow all the time. From the beginning they should grow, and then fall into place, and stay around that number.

**Comment by DavidWiseman** [comment, marked as answer]
It should be possible to do this now with a [Custom Collection](https://dbadash.com/docs/help/custom-collections/) and [Custom Report](https://dbadash.com/docs/how-to/create-custom-reports/).  If you wanted it to appear on the Summary page, a SQL agent job could be configured to fail or a [custom check](https://dbadash.com/docs/help/custom-checks/) could be used.
I might look at adding a collection to track table size and row counts at some point.  The fixed thresholds wouldn't be useful for me, but it might be useful to highlight unusual growths with some configurable thresholds.   Collecting table row counts could be quite heavy on some SQL instances which would need to be considered.  e.g. Databases with a very large number of tables or servers with a large number of databases.

**Reply by AGlennDa** [reply]
Hi David, 
Thank you for your rapid reply. This is much appreciated. 
I did think that somehow this might be possible with a Custom Collection, and Custom Report :) I will try to dig into what a solution would look like for this. Unusual growths are exactly what we need to monitor in a production environment. :)

**Comment by dneilroberts** [comment]
Hi David: I might be interested in monitoring row counts, also. If precise counts aren't required,  in the past I've used sys.dm_db_partition_stats to avoid the heavy cost of count() on very large tables. As long as statistics are updated regularly, it got instant results that were close enough for what I was doing: 

```
SELECT SUM(p.row_count) AS estimated_row_count
FROM sys.dm_db_partition_stats p
JOIN sys.tables t ON p.object_id = t.object_id
WHERE [t.name](http://t.name/) = 'YourTableName' AND p.index_id IN (0, 1);
```

**Comment by DavidWiseman** [comment]
I've been working on this and I'm likely to include it in the next release.

![image](https://github.com/trimble-oss/dba-dash/assets/33157668/ecb8b07b-546e-4341-9c11-7f2967c92c94)

**Reply by AGlennDa** [reply]
Hi David, 
Wow! 👍  
And the ability to see rows per hour, both inserted and dropped are amazing!

**Comment by DavidWiseman** [comment]
This is now available in [3.4](https://dbadash.com/blog/whats-new-in-3.4/).  🚀

The collection is disabled by default for now so you will need to set the schedule in the config tool as described in the link above.
